src/state/slices/marketDataSlice/marketDataSlice.tsx

```diff
@@ -51,32 +52,42 @@ export const defaultMarketData: MarketData = {
   changePercent24Hr: 0,
 }
 
+type CryptoPriceHistoryPayload = { data: HistoryData[]; args: FindPriceHistoryByAssetIdArgs }
+
 export const marketData = createSlice({
   name: 'marketData',
   initialState,
   reducers: {
     clear: () => initialState,
-    setCryptoMarketData: (state, { payload }) => {
-      state.crypto.byId = { ...state.crypto.byId, ...payload } // upsert
-      const ids = Array.from(new Set([...state.crypto.ids, ...Object.keys(payload)]))
-      state.crypto.ids = ids // upsert unique
+    setCryptoMarketData: {
+      reducer: (state, { payload }: { payload: MarketDataById<AssetId> }) => {
+        state.crypto.byId = merge(state.crypto.byId, payload) // upsert
+        // upsert unique
+        state.crypto.ids = Object.keys(state.crypto.byId)
+      },
+
+      // Use the `prepareAutoBatched` utility to automatically
+      // add the `action.meta[SHOULD_AUTOBATCH]` field the enhancer needs
+      prepare: prepareAutoBatched<MarketDataById<AssetId>>(),
     },
-    setCryptoPriceHistory: (
-      state,
-      { payload }: { payload: { data: HistoryData[]; args: FindPriceHistoryByAssetIdArgs } },
```

Comment:
Wouldn't hurt perf to keep `data` and `args` destructured, just a pointer, but potato potato, can't say I find it that much more declarative being destructure